tensorflow_addons/losses/python/focal_loss.py

```diff
@@ -0,0 +1,267 @@
+# Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import abc
+import six
+import math
+import numpy as np
+from tensorflow.python.framework import dtypes
+from tensorflow.python.framework import ops
+from tensorflow.python.framework import smart_cond
+from tensorflow.python.keras.utils.generic_utils import deserialize_keras_object
+from tensorflow.python.keras.utils.generic_utils import serialize_keras_object
+from tensorflow.python.keras.utils.losses_utils import compute_weighted_loss
+from tensorflow.python.keras.utils.tf_utils import is_tensor_or_variable
+from tensorflow.python.ops import array_ops
+from tensorflow.python.ops import check_ops
+from tensorflow.python.ops import math_ops
+from tensorflow.python.ops import nn
+from tensorflow.python.ops.losses import losses_impl
+from tensorflow.python.util.tf_export import keras_export
+from tensorflow.tools.docs import doc_controls
+from tensorflow.python.keras.utils import losses_utils
+
+import tensorflow as tf
+import tensorflow.keras.backend as K
+
+
+@keras_export('keras.losses.Loss')
+class Loss(object):
```

Comment:
Hi, I'm afraid we could use tensorflow.python.keras.losses.LossFunctionWrapper directly. Would you mind referring to https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/python/lifted.py ?

Reply:
@facaiy Sure. I will do that